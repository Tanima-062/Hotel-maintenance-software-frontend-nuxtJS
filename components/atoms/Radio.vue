<template>
  <label class="inline-flex items-center">
    <input
      type="radio"
      :name="name"
      class="form-radio h-4 w-4 mr-1"
      :checked="checked"
      :value="value"
      @click="click(value)"
      :disabled="isDisabled"
    />
    <span :class="`${checked ? 'text-checkboxColor' : color}`">{{ text }}</span>
  </label>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  name!: string

  @Prop({ default: '' })
  value!: string | number

  @Prop({ default: '' })
  text!: string

  @Prop({ default: 'default' })
  color!: string

  @Prop({ default: false })
  checked!: boolean

  @Prop({ default: false })
  isDisabled!: boolean

  @Emit()
  private click(value: string | number): string | number | void {
    if (this.isDisabled) return
    return value
  }
}
</script>
