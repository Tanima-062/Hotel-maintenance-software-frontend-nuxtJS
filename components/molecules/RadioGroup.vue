<template>
  <div>
    <Radio
      v-for="(item, index) in radioItems"
      :key="index"
      :name="name"
      :value="item.value"
      :text="item.text"
      :checked="value === item.value"
      :color="color"
      :isDisabled="isDisabled"
      @click="input"
    />
  </div>
</template>
<script lang="ts">
import { Prop, Vue, Component, Emit } from 'nuxt-property-decorator'
import { Global } from '~/types'

@Component({
  components: {
    Radio: () => import('~/components/atoms/Radio.vue'),
  },
})
export default class extends Vue {
  @Prop()
  name!: string

  @Prop({ default: 'default' })
  color!: string

  @Prop({ default: [] })
  radioItems!: Global.RadioItem[]

  @Prop({ default: false })
  isDisabled!: boolean

  @Prop({ default: '' })
  value!: string | number

  @Emit()
  private input(e: string | number | boolean): string | number | boolean {
    if (this.isDisabled) return ''
    return e
  }
}
</script>
