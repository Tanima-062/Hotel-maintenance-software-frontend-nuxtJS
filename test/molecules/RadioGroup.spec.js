import { mount } from '@vue/test-utils'
import RadioGroup from '@/components/molecules/RadioGroup.vue'
import Radio from '@/components/atoms/Radio.vue'

describe('Molecules RadioGroup Check', () => {
  test('RadioGroup Test', () => {
    const props = {
      radioItems: [
        {
          name: 'test 1',
          value: 'test 1',
        },
        {
          name: 'test 2',
          value: 'test 2',
        },
      ],
    }
    const wrapper = mount(RadioGroup, {
      propsData: props,
      stubs: {
        Radio,
      },
    })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
