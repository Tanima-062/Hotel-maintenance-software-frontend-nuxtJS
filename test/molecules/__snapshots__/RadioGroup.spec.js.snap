// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules RadioGroup Check RadioGroup Test 1`] = `
<div>
  <label
    class="inline-flex items-center"
  >
    <input
      class="form-radio h-4 w-4 mr-1"
      name=""
      type="radio"
      value="test 1"
    />
     
    <span
      class="default"
    >
      
    </span>
  </label>
  <label
    class="inline-flex items-center"
  >
    <input
      class="form-radio h-4 w-4 mr-1"
      name=""
      type="radio"
      value="test 2"
    />
     
    <span
      class="default"
    >
      
    </span>
  </label>
</div>
`;
